flipperdevices/flipperzero-firmware
# NFC: Improvements to NFC Magic app

# What's new

- Ability to write gen1b tags (ignore 0x43)
- Ability to write gen1 7 byte UID tags
- Fix detection of non magic cards

# Verification 

- Try writing gen1b tags, should not fail (no card response to data access command)
- Try writing gen1 7 byte UID tag with 7 byte UID dump (seems only source is https://ikey.ru/22600-mfzero7/, https://i.imgur.com/j51Nu2o.png, thanks to @Foxushka with help in investigation and testing)
- Try writing gen1 4 byte UID tag with 4/7 byte UID dump (regression check), should not be able to write 7 byte dump on it
- Try writing/wiping gen4 (regression check, this time I've tested it, should work from first try :smiley_cat:)
- Try checking any other card (not gen1/gen4), should show "This is wrong card" screen

# Checklist (For Reviewer)
